### PR TITLE
(DOCSP-27231): Flutter tutorial v2

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Update Flutter Device Sync tutorial to be based off of v2 template app.

### Jira

- https://jira.mongodb.org/browse/DOCSP-27231

### Staged Changes

- [Flutter Device Sync Tutorial](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-27231/tutorial/flutter/)

### Reminder Checklist

You might need to also update some corresponding pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-realm update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by dacharyc — APPROVED** [review]
Nice work! Overall, this LGTM and works as written. I've added notes in a couple of spots where you may or may not want to make updates, and there _is_ a file path that needs to be updated before you merge, but aside from that, nothing is blocking.

File: source/tutorial/flutter.txt [hunk 1]
```diff
@@ -145,8 +159,8 @@ Set up the Template App
 
    .. step:: Check the Backend
 
-      Log in to `Atlas App Services <http://cloud.mongodb.com>`_. In the
-      :guilabel:`Atlas` tab, click on :guilabel:`Browse Collections`. In the list
+      Log in to `MongoDB Atlas <http://cloud.mongodb.com>`_. In the
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/tutorial/flutter.txt:162]: oh! huh. yeah we should def doc that. 

tho i'd prefer doing that as a separate cross-sdk unit of work, independent of this PR.

File: source/examples/generated/flutter/tutorial/create_item.snippet.create-item.dart.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     const CreateItemForm({Key? key}) : super(key: key);
+
+     @override
+     createState() => _CreateItemFormState();
+   }
+
+   class _CreateItemFormState extends State<CreateItemForm> {
+     final _formKey = GlobalKey<FormState>();
+     late TextEditingController _itemEditingController;
+     int _priority = PriorityLevel.low;
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/examples/generated/flutter/tutorial/create_item.snippet.create-item.dart.rst:21]: `int` or `var` would both be valid dart.

File: source/examples/generated/flutter/tutorial/todo_item.snippet.todo-item.dart.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     const TodoItem(this.item, {Key? key}) : super(key: key);
+
+     @override
+     Widget build(BuildContext context) {
+       final realmServices = Provider.of<RealmServices>(context);
+       bool isMine = (item.ownerId == realmServices.currentUser?.id);
+       return item.isValid
+           ? ListTile(
+               // ... leading property and child content
+               title: Row(
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/examples/generated/flutter/tutorial/todo_item.snippet.todo-item.dart.rst:21]: i think that the annotated version of the template app will solve this potential issue.

File: source/tutorial/flutter.txt [hunk 4]
```diff
@@ -263,31 +277,24 @@ to only sync Items marked as a High or Severe priority.
 
       The current subscription:
 
-      .. literalinclude:: /examples/generated/flutter/tutorial/init_realm.snippet.base-version.dart
-         :language: dart
-         :caption: lib/realm/init_realm.dart
-         :emphasize-lines: 3-5
+      .. include:: /examples/generated/flutter/tutorial/base-template-subscriptions.rst
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/tutorial/flutter.txt:280]: thx good spot!

File: source/examples/generated/flutter/tutorial/realm_services.snippet.realm-services.dart.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+     bool showAll = false;
+     bool offlineModeOn = false;
+     bool isWaiting = false;
+     late Realm realm;
+     User? currentUser;
+     App app;
+
+     // ... RealmServices initializer
+
+     Future<void> sessionSwitch() async {
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/examples/generated/flutter/tutorial/realm_services.snippet.realm-services.dart.rst:20]: yea, i see what you mean on both points. i think there's some bluehawking funk going on here. investigating.

[the remark this answers is not in the document]
Comment (mongodben) [reply source/examples/generated/flutter/tutorial/realm_services.snippet.realm-services.dart.rst]: figured out. bluehawking funk.

**Pull request merged**